content/post/binding-class-methods-react.md

```diff
@@ -0,0 +1,77 @@
++++
+date = "2016-06-09T13:28:29-07:00"
+draft = false
+title = "Binding es6 class Methods React"
+categories = ["react", "javascript"]
+tags = ["tips", "javascript", "react", "es6"]
+
++++
+
+The choice between using the es6 `class` method for declaring component in
+React seems to be a decided convention within the React community. The
+only issue with it the need to bind the context of `this` everytime you
+create a function. This way you have 
+
+The [Netlify login](https://app.netlify.com/login) form is an easy enough component to look at:
+
+```js
+class LoginPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {loginError: null};
+  }
+
+  onLogin(provider) {
+    return (event) => {
+      ...
+    };
+  }
+
+  render() {
+    const login = () => this.onLogin.bind;
+    return <LoginForm handleLogin={login} loginError={this.state.loginError}/>;
+  }
+}
+```
+Everytime you need to use a class function you need to either bind using
```

Comment:
I would consider maybe not inlining the code here. Something more like? 

Everytime you need to use a class function you have one of two options:
```
() => this.onLogin
```
The problem is that this.....
The other options is
``` 
this.onLogin.bind(this)
```
Which seems less hacky, but requires declaring a new function.